src/service/routes/users.js

```diff
@@ -29,4 +36,68 @@ router.get('/:id', async (req, res) => {
   res.send(user);
 });
 
+// Add SSH public key
+router.post('/:username/ssh-keys', async (req, res) => {
+  if (!req.user) {
+    res.status(401).json({ error: 'Authentication required' });
+    return;
+  }
+
+  const targetUsername = req.params.username.toLowerCase();
+
+  // Only allow users to add keys to their own account, or admins to add to any account
+  if (req.user.username !== targetUsername && !req.user.admin) {
+    res.status(403).json({ error: 'Not authorized to add keys for this user' });
+    return;
+  }
+
+  const { publicKey } = req.body;
+  if (!publicKey) {
+    res.status(400).json({ error: 'Public key is required' });
+    return;
+  }
+
+  // Strip the comment from the key (everything after the last space)
+  const keyWithoutComment = publicKey.split(' ').slice(0, 2).join(' ');
```

Comment:
Should we add some validation here to make sure that the key is in a valid format? Or would this be more appropriate for the db functions?

Right now, invalid formatting would get caught in the try/catch below, but the user-facing error wouldn't be descriptive.